shim: compile error with _Generic and MSVC 2019

The compiler used is running on Windows 10.
The compiler is from MSVC 2019 but not the latest version within that package.

```
\src\shim\shim.c(50): warning C4013: '_Generic' undefined; assuming extern returning int
\src\shim\shim.c(50): error C2275: 'uint16_t': illegal use of this type as an expression
```


The compiler does not understand the keyword `_Generic`. It would be good to make the code
compile also for compilers that does not understand _Generic.

--- a/include/re_atomic.h
+++ b/include/re_atomic.h
@@ -8,7 +8,8 @@
 #define RE_H_ATOMIC__
 
 /* C11 */
-#if defined(HAVE_ATOMIC) && !defined(__STDC_NO_ATOMICS__)
+#if defined(HAVE_ATOMIC) && __STDC_VERSION__ >= 201112L &&                    \
+	!defined(__STDC_NO_ATOMICS__)
 
 #include <stdatomic.h>
 

--- a/include/re_convert.h
+++ b/include/re_convert.h
@@ -55,6 +55,7 @@ static inline int try_into_err(void *dest, ...)
 }
 
 
+#if __STDC_VERSION__ >= 201112L /* Needs C11 support */
 /**
  * Try to convert safely from one type (src) into another (dest).
  * Types are auto detected.
@@ -77,3 +78,4 @@ static inline int try_into_err(void *dest, ...)
 			)						      \
 		)							      \
 	(&(dest), (src))
+#endif

--- a/src/shim/shim.c
+++ b/src/shim/shim.c
@@ -47,7 +47,7 @@ static bool shim_send_handler(int *err, struct mbuf *mb, void *arg)
 		return true;
 	}
 
-	err_len = try_into(len, mbuf_get_left(mb));
+	err_len = try_into_u16_from_size(&len, mbuf_get_left(mb));
 	if (err_len) {
 		DEBUG_WARNING("send: mbuf to big\n");
 		*err = err_len;

